File: app/src/main/java/com/streamarc/tv/data/Service.kt
package com.streamarc.tv.data

import com.streamarc.tv.BuildConfig

/**
 * The two content services the app connects to. Each has its own
 * Xtream Codes server (supplied at build time) and its own sign-in.
 */
enum class Service(val title: String, val baseUrl: String, val kind: ContentKind) {
    LIVE("Live TV", BuildConfig.LIVE_URL.trim().trimEnd('/'), ContentKind.LIVE),
    VOD("Video on Demand", BuildConfig.VOD_URL.trim().trimEnd('/'), ContentKind.MOVIE);

    /** False when this build was made without a server address for the service. */
    val isConfigured: Boolean get() = baseUrl.isNotBlank()
}

enum class ContentKind { LIVE, MOVIE }

File: macos/StreamArcTV/Data/Service.cs
namespace StreamArcTV.Data;

/// <summary>
/// The two content services the app connects to. Each has its own Xtream Codes server
/// (supplied at build time) and its own sign-in.
/// </summary>
public enum Service { LIVE, VOD }

public enum ContentKind { LIVE, MOVIE, SERIES }

public static class ServiceInfo
{
    public static string Title(this Service s) => s == Service.LIVE ? "Live TV" : "Video on Demand";

    public static string BaseUrl(this Service s) =>
        (s == Service.LIVE ? BuildInfo.LiveUrl : BuildInfo.VodUrl).Trim().TrimEnd('/');

    public static ContentKind Kind(this Service s) => s == Service.LIVE ? ContentKind.LIVE : ContentKind.MOVIE;

    /// False when this build was made without a server address for the service.
    public static bool IsConfigured(this Service s) => !string.IsNullOrWhiteSpace(s.BaseUrl());

    public static readonly Service[] All = { Service.LIVE, Service.VOD };
}
